### PR TITLE
Add tool to build Demo

### PR DESCRIPTION
**Pull request by navinns**
https://master-navinns-497220694.us-west-2.elb.amazonaws.com/job/csdk/job/pr_buildcheck/1/console

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tools/ci/ci_demos.py [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+
+
+def _check_status(result):
+    print("\n----------------------------------------------------------------")
+    print("Build Result")
+    print("----------------------------------------------------------------")
+    print(yaml.dump(result))
+
+    status = any([_k for _k, _v in result.items() if _v.get("status") == "FAIL"])
+    if status:
```
Comment (pavanmr94) [inline tools/ci/ci_demos.py:191]: might want to dump an xml (junit)

File: tools/ci/ci_demos.py [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+        shutil.rmtree(dir_name)
+    except OSError as err:
+        print("Error: %s - %s." % (err.filename, err.strerror))
+
+
+def _read_config(_path):
+    try:
+        _config = yaml.load(_path.read_text())
+        return _config
+    except:
```
Comment (pavanmr94) [inline tools/ci/ci_demos.py:166]: bare except. Might want to use `except Exception:`

File: tools/ci/ci_demos.py [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+
+def cli_build(args):
+    _file = Path(args.config_file)
+    _config = _read_config(_file)
+    _build_flags = _config.get("_default", {}).get("build_flags", [])
+
+    _config_build(args.src, _build_flags, args.build_path)
+    _targets = _get_targets(args.build_path)
+
+    allowed = "|".join(_config.get("_default", {}).get("allow", []))
```
[the remark this answers is not in the document]
Comment (navinns) [reply tools/ci/ci_demos.py:41]: format is not fixed for this one yet

File: tools/ci/ci_demos.py (hunk 1 above)
Comment (navinns) [reply tools/ci/ci_demos.py:191]: will add

**Pull request merged**